### ISSUE TITLE
RMSE by MeanSquaredError(squared=False) not correctly when using MetricTracker

### ISSUE THREAD

**Issue by huidiwiwi** [Lightning-AI/metrics#1109]
## 🐛 Bug

The tracked rmse metric is different from other metrics, and not the same as the normal rmse. rmse coincides with the sqrt(mse) or sqrt(mse_tracked)  
### To Reproduce
Steps to reproduce the behavior...
in a litmoudle class: 
initial the metrics:
``` py
metrics = MetricCollection(
          {
              "mse": MeanSquaredError(),
              "rmse": MeanSquaredError(squared=False),
          }
      )
self.val_metrics = metrics.clone(prefix="val/")
self.val_metric_tracker = MetricTracker(metrics.clone(prefix="val/", postfix = "_track"), maximize= len(metrics) * [False])
```
log the metrics using pytorch lightning hoods:
```py
def on_validation_epoch_start(self):
    self.val_metric_tracker.increment()

def validation_step(self, batch: Any, batch_idx: int):
    loss, preds, targets = self.step(batch)
    self.log("val/loss", loss, on_step=True, on_epoch=True, prog_bar=True)

    self.val_metrics(preds, targets)
    self.val_metric_tracker.update(preds, targets)

    self.log_dict(self.val_metrics, on_step=False, on_epoch=True, prog_bar=True)
    self.log_dict(self.val_metric_tracker.compute(), on_step=True, on_epoch=False, prog_bar=True)
    return {"loss": loss, "preds": preds, "targets": targets}

def validation_epoch_end(self, outputs: List[Any]):
    best_metrics, epoch_info = self.val_metric_tracker.best_metric(return_step=True)
    best_val = {str(key) + '_best': val for key, val in best_metrics.items()}
    self.log_dict(best_val, on_step=False, on_epoch=True, prog_bar=True)

```

### Expected behavior

RMSE metrics under tracker should give a near identical value as RMSE without MetricTracker which coincides with the sqrt(mse)
RMSE under tracker should behavior the seem as MSE, at least the final best_metric() should return the correct value

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source):
- Python & PyTorch Version (e.g., 1.0): torch 1.11.0
- Pytorch Lightning Version: 1.6.3
- mlflow version: 1.26.1
- Any other relevant information such as OS (e.g., Linux): Win

### Additional context

It seems like the MetricTracker only track the first RMSE at the begining step of every epoch but not updating it. 
example output in my use case:

| MSE /epoch       | val/mse (epoch)  | val/mse_track (step)  | val/mse_track_best (epoch)|
| ------------- |:-------------:| -----:| -----:|
| 0  | 534882 | [70590, 576224,6293749 .... ,534882] | 534882|
| 1  | 507540 | [...,507540]   | 507540|
| 2  | 501275 | [...,501275] | 501275|
| 3  | 502213 | [...,502213] | 501275|

| RMSE        | val/rmse (epoch)  | val/rmse_track (step)  | val/rmse_track_best (epoch)|
| ------------- |:-------------:| -----:| -----:|
| 0  | 231=sqrt(534882) | [265, 265,265 .... ,265] | 265|
| 1  | 225 | [259,...,259]   | 259|
| 2  | 223 | [258,...,258] | 258|
| 3  | 228 | [258,...,258] | 258|

compare mae with mae loss, we see the tracker will update the metirc in each step by average at online (72^=(82+61)/2), and the best_meric=min( metric_track) should update the min value of metric_epoch.
| MAE/L1Loss | val/mae = val/loss (epoch)   | val/mae_track (step)| val/loss(step)  | val/mae_track_best (epoch)|
| -------------  |:-------------:| -----:| -----:| -----:|
| 0  | 70 | [82, 72,76 .... ,70] | [82, 61,85...,64]| 70|
| 1  | 69 | [83,72,...,69]   | [83,60,...,64]| 69|
|...|

**Comment by SkafteNicki** [comment]
Hi @huidiwiwi, 
Could you try initializing the metrics in this way:
```python
metrics = MetricCollection(
          {
              "mse": MeanSquaredError(),
              "rmse": MeanSquaredError(squared=False),
          }
          compute_groups=False,
      )
```
if that solves the problem, it is not related to the metric tracker but instead the metric collection. I recently fixes some bugs related to this (https://github.com/Lightning-AI/metrics/pull/1076) so you can also try installing from master and see if that helps.

**Comment by huidiwiwi** [comment]
> Hi @huidiwiwi, Could you try initializing the metrics in this way:
> 
> ```python
> metrics = MetricCollection(
>           {
>               "mse": MeanSquaredError(),
>               "rmse": MeanSquaredError(squared=False),
>           }
>           compute_groups=False,
>       )
> ```
> 
> if that solves the problem, it is not related to the metric tracker but instead the metric collection. I recently fixes some bugs related to this (#1076) so you can also try installing from master and see if that helps.

Hi @SkafteNicki , thanks for your help. Yes, I tested with compute_groups=False, it updates now as expected. I will test the new master version later. 
But the MetricTracker still not working perfectly as the sanity check will be tracked as well. I tried to reset(/reset_all) the tracker under the hook on_train_start, but it keeps giving then nans or 0 at each val end. Would be glad to see if you have any idea about it. thanks in advance

**Comment by SkafteNicki** [comment]
> But the MetricTracker still not working perfectly as the sanity check will be tracked as well. I tried to reset(/reset_all) the tracker under the hook on_train_start, but it keeps giving then nans or 0 at each val end. Would be glad to see if you have any idea about it. thanks in advance

I know this is the stupid solution, but you could disable sanity checking. That said, I would recommend you to call `self.val_metric_tracker.reset()` in the `on_train_start` hook, which should reset whatever was stored during sanity checking. I do not understand why this is not working for you as it should only be called once in the beginning.

**Comment by SkafteNicki** [comment]
Here is a solution to the sanity checking problem:
```python
  def __init__(self):
      super().__init__()
      ...
      self.counter = 0 

  def on_validation_epoch_start(self):
      if self.counter != 1:
          self.val_metric_tracker.increment()
      else:
          self.val_metric_tracker.reset()
      self.counter += 1
```
the counter will only reset after the sanity checking is done, else it will increase the tracker.
Closing issue.
